app/models/concerns/operations/order_fulfilment_operations.rb

```diff
@@ -82,6 +82,9 @@ def dispatch(ord_pkg, quantity:, from_location:)
           source: ord_pkg,
           location: location
         )
+
+        ord_pkg.order.start_dispatching if ord_pkg.order.awaiting_dispatch?
```

Comment:
Why we are adding a check `ord_pkg.order.awaiting_dispatch?`.

state_machine handles it on its own. If we try to perform operation which is not allowed from specific states then its throws error.

As per my understanding order can be transitioned to `dispatching` state only from `closed` and `awaiting_dispatched` state.

Do we want to avoid a transition from closed to dispatching state here? Because we cant dispatch items to closed order I think.

Or maybe just added for safety purposes. 

Could you please help me understand? 

Thanks in advance. 😉

Reply:
We only want to start processing if it's in `awaiting_dispatch` state.

Even if the state_machine handles bad cases, I don't rely on it.

Plus the extra `if` adds context to the statement. By reading it we understand that we *can* dispatch an orders_package from a scheduled order.

Even if you _can_ remove a statement, doesn't always mean you _should_